File: Recursion/Sum of First N Natural Numbers/SumOfFirstN.py
number = int(input("Enter the limit: "))

def sumOfFirstN(n):
    if n == 1:
        return 1
    return n + sumOfFirstN(n-1)

if number < 0:
    print("Error")
else:
    print("Sum of first ", number, " number is: ", sumOfFirstN(number))

File: Recursion/002.Sum of First N Natural Numbers/SumOfFirstN.swift
//
//  SumOfFirstN
//
//  Created by Feialoh Francis on 23/11/22.
//

class SumOfFirstN {
    
    var num: Int64
    
    init(num: Int64) {
        self.num = num
    }
    
    func sumOfN(_ n: Int64) -> Int64 {
        
        if n == 1 {
            return 1
        }
        return n+sumOfN(n-1)
    }
}

print("Enter the limit: ")
let num = Int64(readLine() ?? "0")

if let num = num {
    let sol = SumOfFirstN(num: num)
    print("Sum of first \(num) number is = \(sol.sumOfN(num))")
}
